actionmailbox/app/controllers/action_mailbox/ingresses/amazon_ses/sns_notification.rb

```diff
@@ -0,0 +1,74 @@
+# frozen_string_literal: true
+
+module ActionMailbox
+  module Ingresses
+    module AmazonSes
+      class SnsNotification
+        class MessageContentError < StandardError; end
+
+        def initialize(request_body)
+          require "aws-sdk-sns"
```

Comment:
Can't we move this require to the top of the file? We also probably need to raise an error when this gem isn't present.

Reply:
Done - based (loosely) on ActiveRecord adapter loader:
```ruby
begin
  require "aws-sdk-sns"
rescue LoadError
  raise LoadError,
        "Could not load the 'aws-sdk-sns' gem. Ensure that you've added the gem to your Gemfile.", e.backtrace
end
```
Let me know if you want this adjusted.